rust/src/delta.rs

```diff
@@ -1130,10 +1151,16 @@ impl<'a> DeltaTransaction<'a> {
     async fn try_commit(
         &mut self,
         tmp_log_path: &str,
+        version: Option<DeltaDataTypeVersion>,
```

Comment:
I think it's worth making version non-optional here, that way the commit_with could just do:

* Fetch the next attempt version
* Call commit_version

That way the bulk of the work goes into commit_version. That may also obviate the need for the  `log_bytes_with_actions` function as well

Reply:
FYI @rtyler - I implemented your suggestion to make `version` non-optional in this method and the implication that `commit_version` doesn't need the optimistic concurrency loop at all and also fixed naming of `log_bytes_from_actions` (it doesnt return bytes lol). However, rather than funnel `commit_with` down to a call of `commit_version`, I left the separation in place so that call stacks now look like:

```
commit_version
log_entry_from_actions
try_commit
```

```
commit_with
log_entry_from_actions
try_commit_loop
try_commit
```

My reasoning here is that `try_commit_loop` shouldn't have to serialize the  same set of log actions on every iteration of the loop until succeeding, so rather than re-use `commit_version`, I think its better if both `commit_version` and `commit_with` serialize the log entry independently and funnel down separate paths to the `try_commit` call.